### DISCUSSION TITLE
How do I update the query string?

### DISCUSSION THREAD

**Discussion by ojii** [leptos-rs/leptos#2692] [Q&A] [answered]
I know I can get the current query string values from `use_query`/`use_query_map`, but how do I update the query string client side? I'd like to track the value of an input field in the URL so they can be shared.

**Comment by gbj** [comment, marked as answer]
[Here's an example](https://github.com/leptos-rs/leptos/blob/main/examples/counter_url_query/src/lib.rs) of doing exactly that. Otherwise you can just `use_navigate` and navigate to a URL with the new query string.

**Reply by ojii** [reply]
thank you. i should've looked at all the examples, sorry.
